src/api/models/base_model_.py

```diff
@@ -0,0 +1,80 @@
+"""API Models Logic."""
+
+# Standard Python Libraries
+import pprint
+import typing
+
+# Third-Party Libraries
+import six  # type: ignore
+
+# cisagov Libraries
+from api import util
+
+CustomType = typing.TypeVar("CustomType")
+
+
+class Model:
+    """Mode Class."""
+
+    # swaggerTypes: The key is attribute name and the
+    # value is attribute type.
+    # example = {"key": "val"}
+
+    swagger_types = {}  # type: ignore
+
+    # attributeMap: The key is attribute name and the
+    # value is json key in definition.
+    attribute_map = {}  # type: ignore
+
+    @classmethod
+    def from_dict(cls: typing.Type[CustomType], dikt) -> CustomType:
+        """Return the dict as a model."""
+        return util.deserialize_model(dikt, cls)
+
+    def to_dict(self):
+        """Return the model properties as a dict.
+
+        :rtype: dict
+        """
+        result = {}
+
+        for attr, _ in six.iteritems(self.swagger_types):
+            value = getattr(self, attr)
+            if isinstance(value, list):
+                result[attr] = list(
+                    map(lambda x: x.to_dict() if hasattr(x, "to_dict") else x, value)
+                )
+            elif hasattr(value, "to_dict"):
+                result[attr] = value.to_dict()
+            elif isinstance(value, dict):
+                result[attr] = dict(
+                    map(
+                        lambda item: (item[0], item[1].to_dict())
+                        if hasattr(item[1], "to_dict")
+                        else item,
+                        value.items(),
+                    )
+                )
+            else:
+                result[attr] = value
+
+        return result
+
+    def to_str(self):
+        """Return the string representation of the model.
+
+        :rtype: str
+        """
+        return pprint.pformat(self.to_dict())
+
+    def __repr__(self):
+        """For `print` and `pprint`."""
+        return self.to_str()
+
+    def __eq__(self, other):
+        """Return true if both objects are equal."""
+        return self.__dict__ == other.__dict__
+
+    def __ne__(self, other):
+        """Return true if both objects are not equal."""
+        return not self == other
```

Comment:
Why is `__eq__` checking for equality of the `__dict__` values, but `__ne__` isn't?  I'd expect that these functions would be identical except for the `not` before the comparison.

Reply:
I'm assuming it's because of Python's [data model documentation](https://docs.python.org/3.10/reference/datamodel.html#object.__lt__) leading to writing `__ne__()` as a negation of `__eq__()`. This is actually the default behavior I believe per the linked documentation:
> For [`__ne__()`](https://docs.python.org/3.10/reference/datamodel.html#object.__ne__), by default it delegates to [`__eq__()`](https://docs.python.org/3.10/reference/datamodel.html#object.__eq__) and inverts the result unless it is `NotImplemented`.

Reply:
Ah, thanks.  The part that didn't click in my brain is that `not self == other` is actually `not self.__eq__(other)`.  This makes sense to me now.